tsconfig.json

```diff
@@ -3,20 +3,18 @@
     "outDir": "./dist/",
     "sourceMap": true,
     "noImplicitAny": true,
-    "module": "commonjs",
-    "target": "es5",
+    "module": "esnext",
```

Comment:
So with this change, tsc will now generate code with `import` statements in it?
And then webpack now compiles this into one js file, right?
I don't see webpack changes, does it handle this automatically?

Also I remember having to do something special for libraries that were built with esnext since we were mixing them with our commonjs code imports.  Hopefully those things wouldn't be needed if tsc is compiling to esnext too. 
But, I'd guess if we use this new configuration we'll have to deal with this if we are importing a new package based on this config into one of our legacy projects we'll have to deal with those issues.

Moving to esnext seems great to me, I'm curious about how all of this works.

Reply:
Yes that all sounds good.